commands/container-registry/repository/list.go

```diff
@@ -0,0 +1,138 @@
+package repository
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fatih/structs"
+	"github.com/ionos-cloud/ionosctl/v6/commands/cloudapi-v6/query"
+	"github.com/ionos-cloud/ionosctl/v6/commands/container-registry/registry"
+	"github.com/ionos-cloud/ionosctl/v6/internal/client"
+	"github.com/ionos-cloud/ionosctl/v6/internal/constants"
+	"github.com/ionos-cloud/ionosctl/v6/internal/core"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/json2table/jsonpaths"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/jsontabwriter"
+	"github.com/ionos-cloud/ionosctl/v6/internal/printer/tabheaders"
+	cloudapiv6 "github.com/ionos-cloud/ionosctl/v6/services/cloudapi-v6"
+	"github.com/ionos-cloud/ionosctl/v6/services/cloudapi-v6/resources"
+	ionoscloud "github.com/ionos-cloud/sdk-go-container-registry"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func RepositoryListCmd() *core.Command {
+	c := core.NewCommand(
+		context.Background(), nil, core.CommandBuilder{
+			Namespace:  "container-registry",
+			Resource:   "repository",
+			Verb:       "list",
+			Aliases:    []string{"ls", "l"},
+			ShortDesc:  "Retrieve all repositories.",
+			LongDesc:   "Retrieve all repositories in a registry.",
+			Example:    "ionosctl container-registry list",
+			PreCmdRun:  PreCmdList,
+			CmdRun:     CmdList,
+			InitClient: true,
+		},
+	)
+	c.Command.Flags().StringSlice(constants.ArgCols, defaultCols, tabheaders.ColsMessage(allCols))
+	_ = c.Command.RegisterFlagCompletionFunc(
+		constants.ArgCols,
+		func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return allCols, cobra.ShellCompDirectiveNoFileComp
+		},
+	)
+
+	c.AddStringFlag(constants.FlagRegistryId, constants.FlagRegistryIdShort, "", "Registry ID")
+	_ = c.Command.RegisterFlagCompletionFunc(
+		constants.FlagRegistryId,
+		func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return registry.RegsIds(), cobra.ShellCompDirectiveNoFileComp
+		},
+	)
+
+	c.AddSetFlag(
+		cloudapiv6.ArgOrderBy, "", "-lastPush", []string{
+			"-lastPush", "-lastPull", "-artifactCount", "-pullCount", "-pushCount", "name", "lastPush",
```

Comment:
Are these minuses intended? `-lastPull` and `lastPull`? does this order them in reverse order?

Reply:
Yes, the minuses are intended and they do exactly what you assumed. When used without minus, the results are in ascending order and with the minus, they are in descending order.